### ISSUE TITLE
'Automatic links' are nuked

### ISSUE THREAD

**Issue by kmac** [tj/mad#24]
I notice that markdown [automatic links](http://daringfireball.net/projects/markdown/syntax#autolink) are completely nuked by the expression on line 90: `s|<(.+?)>||g; \`

e.g. if I have something like this in my source:

    See <http://someurl.com/>

It is rendered with the link comletely missing. As a workaround I would need to write:

    See [http://someurl.com/](<http://someurl.com/>)

which is cumbersome and doesn't look as nice.

Note that in my case I don't want a plaintext URL because it's not recognized when i convert the same document to HTML.

Basically, I'm wondering if there's a good reason for this statement at all? It will generally strip out URLs which are usually desired to see in the text form especially.

thanks!

**Comment by ernstki** [comment]
Maybe what TJ was going for was to strip HTML tags, including `<!-- HTML comments -->`? He probably wasn't trying to implement the whole Markdown spec.

You can have the best of both worlds, though, with a one-line change to `bin/mad`.

I liked the link text highlighting of PR #22, so I just added another Perl `s///` expression to highlight the automatic links in angle brackets just like normal Markdown links, and that looks [like this](https://github.com/tj/mad/compare/master...ernstki:master#diff-1039e0dff895cac32de3b7ae76fca82cR106):

```perl
      ⋮
      # line 90-ish of bin/mad
      s^<((ftp|http)s?://.+?)>^\e[$em\e[$link\1\e[0m^g; \ 
      ⋮
```

This isn't a straight copy-paste solution, though; you'll need to modify that slightly to your liking, since PR #22 adds a new `link` style to the config file, which you probably won't have unless you've applied that [diff](https://patch-diff.githubusercontent.com/raw/tj/mad/pull/22.diff) to your copy of `mad`.
